cpp/source/deal.II/supercapacitor.templates.h

```diff
@@ -39,8 +40,12 @@ void SuperCapacitorInspector<dim>::inspect(EnergyStorageDevice *device)
   static int i = 0;
   SuperCapacitor<dim> *supercapacitor =
       dynamic_cast<SuperCapacitor<dim> *>(device);
-  BOOST_ASSERT_MSG(supercapacitor != nullptr,
-                   "There was a problem casting the device pointer.");
+  if (supercapacitor == nullptr)
+  {
+    std::bad_cast exception;
+    throw exception;
```

Comment:
`throw std::bad_cast();`
Please comment and explain why you had to throw the exception yourself.